guides/creating-a-custom-stepper-using-the-cdk-stepper.md

```diff
@@ -0,0 +1,187 @@
+
+```typescript
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { CdkStepperModule } from '@angular/cdk/stepper'; // this is the relevant import
+
+import { AppComponent } from './app.component';
+
+@NgModule({
+  imports: [BrowserModule, CdkStepperModule], // add the module to your imports
+  declarations: [AppComponent],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+## Create our custom stepper component
+
+Now we are ready to create our custom stepper component. Therefore, we need to create a new Angular component which extends `CdkStepper`:
+
+__custom-stepper.component.ts__
+```typescript
+import { Directionality } from '@angular/cdk/bidi';
+import { ChangeDetectorRef, Component, Input, QueryList } from '@angular/core';
+import { CdkStep, CdkStepper } from '@angular/cdk/stepper';
+
+@Component({
+  selector: 'app-custom-stepper',
+  templateUrl: './custom-stepper.component.html',
+  styleUrls: ['./custom-stepper.component.css'],
+  providers: [{ provide: CdkStepper, useExisting: CustomStepperComponent }],
+})
+export class CustomStepperComponent extends CdkStepper {
+  /** Whether the validity of previous steps should be checked or not */
+  linear: boolean;
+
+  /** The index of the selected step. */
+  selectedIndex: number;
+
+  /** The list of step components that the stepper is holding. */
+  steps: QueryList<CdkStep>;
+
+  constructor(dir: Directionality, changeDetectorRef: ChangeDetectorRef) {
+    super(dir, changeDetectorRef);
+  }
+
+  onClick(index: number): void {
+    this.selectedIndex = index;
+  }
+}
+```
+
+After we've extended our component class from `CdkStepper` we can now access different properties from this class like `linear`, `selectedIndex` and `steps` which are defined in the [API documentation](https://material.angular.io/cdk/stepper/api#CdkStepper). 
+
+This is the HTML template of our custom stepper component:
+
+__custom-stepper.component.html__
+```html
+<section class="container">
+    <header>
+      <h2>Step {{selectedIndex + 1}}/{{steps.length}}</h2>
+    </header>
+  
+    <section *ngFor="let step of steps; let i = index;">
+      <div [style.display]="selectedIndex === i ? 'block' : 'none'">
+          <!-- This is where the content from each CdkStep is projected to -->
+          <ng-container [ngTemplateOutlet]="step.content"></ng-container>
+        </div>
+    </section>
+  
+    <footer class="step-navigation-bar">
+      <button class="nav-button" cdkStepperPrevious>&larr;</button>
+      <button class="step" *ngFor="let step of steps; let i = index;" [ngClass]="{'active': selectedIndex === i}" (click)="onClick(i)">Step {{i + 1}}</button>
+      <button class="nav-button" cdkStepperNext>&rarr;</button>
+    </footer>
+</section>
+```
+
+In the `app.component.css` file we can now style the stepper however we want:
+
+__custom-stepper.component.css__
+```css
+.container {
```

Comment:
We always namespace all CSS, even in examples

Reply:
Done